packages/openapi-fetch/src/index.ts

```diff
@@ -19,55 +21,24 @@ interface ClientOptions extends RequestInit {
+export type QuerySerializer<T> = (query: T extends { parameters: any } ? NonNullable<T["parameters"]["query"]> : Record<string, unknown>) => string;
+export type BodySerializer<T> = (body: OperationRequestBodyContent<T>) => any;
 export type ParseAs = "json" | "text" | "blob" | "arrayBuffer" | "stream";
-export interface OperationObject {
-  parameters: any;
-  requestBody: any; // note: "any" will get overridden in inference
-  responses: any;
+export interface DefaultParamsOption {
+  params?: { query?: Record<string, unknown> };
 }
-export type HttpMethod = "get" | "put" | "post" | "delete" | "options" | "head" | "patch" | "trace";
-export type OkStatus = 200 | 201 | 202 | 203 | 204 | 206 | 207 | "2XX";
-// prettier-ignore
-export type ErrorStatus = 500 | '5XX' | 400 | 401 | 402 | 403 | 404 | 405 | 406 | 407 | 408 | 409 | 410 | 411 | 412 | 413 | 414 | 415 | 416 | 417 | 418 | 420 | 421 | 422 | 423 | 424 | 425 | 426 | 429 | 431 | 444 | 450 | 451 | 497 | 498 | 499 | '4XX' | "default";
-
-// util
-/** Get a union of paths which have method */
-export type PathsWith<Paths extends Record<string, PathItemObject>, PathnameMethod extends HttpMethod> = {
-  [Pathname in keyof Paths]: Paths[Pathname] extends { [K in PathnameMethod]: any } ? Pathname : never;
-}[keyof Paths];
-/** Find first match of multiple keys */
-export type FilterKeys<Obj, Matchers> = { [K in keyof Obj]: K extends Matchers ? Obj[K] : never }[keyof Obj];
-export type MediaType = `${string}/${string}`;
-
-// general purpose types
-export type Params<T> = T extends { parameters: any } ? { params: NonNullable<T["parameters"]> } : BaseParams;
-export type RequestBodyObj<T> = T extends { requestBody?: any } ? T["requestBody"] : never;
-export type RequestBodyContent<T> = undefined extends RequestBodyObj<T> ? FilterKeys<NonNullable<RequestBodyObj<T>>, "content"> | undefined : FilterKeys<RequestBodyObj<T>, "content">;
-export type RequestBodyMedia<T> = FilterKeys<RequestBodyContent<T>, MediaType> extends never ? FilterKeys<NonNullable<RequestBodyContent<T>>, MediaType> | undefined : FilterKeys<RequestBodyContent<T>, MediaType>;
-export type RequestBody<T> = RequestBodyMedia<T> extends never ? { body?: never } : undefined extends RequestBodyMedia<T> ? { body?: RequestBodyMedia<T> } : { body: RequestBodyMedia<T> };
-export type QuerySerializer<T> = (query: T extends { parameters: any } ? NonNullable<T["parameters"]["query"]> : Record<string, unknown>) => string;
-export type BodySerializer<T> = (body: RequestBodyMedia<T>) => any;
-export type RequestOptions<T> = Params<T> &
-  RequestBody<T> & {
+export type ParamsOption<T> = T extends { parameters: any } ? { params: NonNullable<T["parameters"]> } : DefaultParamsOption;
+export type RequestBodyOption<T> = OperationRequestBodyContent<T> extends never ? { body?: never } : undefined extends OperationRequestBodyContent<T> ? { body?: OperationRequestBodyContent<T> } : { body: OperationRequestBodyContent<T> };
```

Comment:
The types that were left in here are opinionated and meant specifically for `openapi-fetch`. They are still exported so they can still be used. But is more of an explicit about “you’re probably only importing this because you want to mimic / mock `openapi-fetch` behavior specifically” while leaving the generic usecases to the new types package